### PR TITLE
BREAKING CHANGE(@formatjs/intl-numberformat): ES2021 compliant

### PR DESCRIPTION
**Pull request by pyrocat101**
BREAKING CHANGE(@formatjs/intl-numberformat): ES2021 compatible Intl.NumberFormat.

This is a rewrite of Intl.NumberFormat polyfill:
- Feature compatible with ES2021 spec.
- Pass most of Intl.NumberFormat test262 cases.
- New version is 3 to 4 times faster in the benchmark of instantiating NumberFormat object + formatting numbers in currency or unit styles. It is almost 2 times faster in the benchmark variant that reuses the NumberFormat object.
- Slightly reduce locale data JSON size.
- Support compound unit like `kilometer-per-second`, where `kilometer` and `second` can be any simple unit whitelisted in [IsSanctionedSimpleUnitIdentifier][].
- Polyfill for `NumberFormat.prototype.toLocaleString`.
- Drop `Unified` prefix from the type definitions.
- Reorganized source module structure.

BREAKING CHANGE(formatjs-extract-cldr-data): Extracted CLDR unit data has breaking schema change.

BREAKING CHANGE(@formatjs/intl-utils): most of NumberFormat public and internal types are moved out of this package.

[IsSanctionedSimpleUnitIdentifier]: https://tc39.es/ecma402/#sec-issanctionedsimpleunitidentifier

### CODE REVIEW

File: packages/intl-numberformat/src/data/currency-digits.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"ADP":0,"AFN":0,"ALL":0,"AMD":2,"BHD":3,"BIF":0,"BYN":2,"BYR":0,"CAD":2,"CHF":2,"CLF":4,"CLP":0,"COP":2,"CRC":2,"CZK":2,"DEFAULT":2,"DJF":0,"DKK":2,"ESP":0,"GNF":0,"GYD":2,"HUF":2,"IDR":2,"IQD":0,"IRR":0,"ISK":0,"ITL":0,"JOD":3,"JPY":0,"KMF":0,"KPW":0,"KRW":0,"KWD":3,"LAK":0,"LBP":0,"LUF":0,"LYD":3,"MGA":0,"MGF":0,"MMK":0,"MNT":2,"MRO":0,"MUR":2,"NOK":2,"OMR":3,"PKR":2,"PYG":0,"RSD":0,"RWF":0,"SEK":2,"SLL":0,"SOS":0,"STD":0,"SYP":0,"TMM":0,"TND":3,"TRL":0,"TWD":2,"TZS":2,"UGX":0,"UYI":0,"UYW":4,"UZS":2,"VEF":2,"VND":0,"VUV":0,"XAF":0,"XOF":0,"XPF":0,"YER":0,"ZMK":0,"ZWD":0}
```
Comment (longlho) [inline packages/intl-numberformat/src/data/currency-digits.json:1]: can u add `@generated` tag & `prettier-ignore` here?

File: packages/intl-numberformat/src/data/digit-mapping.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"arab":["٠","١","٢","٣","٤","٥","٦","٧","٨","٩"],"arabext":["۰","۱","۲","۳","۴","۵","۶","۷","۸","۹"],"bali":["୐","୑","୒","୓","୔","୕","ୖ","ୗ","୘","୙"],"beng":["০","১","২","৩","৪","৫","৬","৭","৮","৯"],"deva":["०","१","२","३","४","५","६","७","८","९"],"fullwide":["༐","༑","༒","༓","༔","༕","༖","༗","༘","༙"],"gujr":["૦","૧","૨","૩","૪","૫","૬","૭","૮","૯"],"guru":["੦","੧","੨","੩","੪","੫","੬","੭","੮","੯"],"khmr":["ߠ","ߡ","ߢ","ߣ","ߤ","ߥ","ߦ","ߧ","ߨ","ߩ"],"knda":["೦","೧","೨","೩","೪","೫","೬","೭","೮","೯"],"laoo":["໐","໑","໒","໓","໔","໕","໖","໗","໘","໙"],"limb":["ॆ","े","ै","ॉ","ॊ","ो","ौ","्","ॎ","ॏ"],"mlym":["൦","൧","൨","൩","൪","൫","൬","൭","൮","൯"],"mong":["ࠐ","ࠑ","ࠒ","ࠓ","ࠔ","ࠕ","ࠖ","ࠗ","࠘","࠙"],"mymr":["@","A","B","C","D","E","F","G","H","I"],"orya":["୦","୧","୨","୩","୪","୫","୬","୭","୮","୯"],"tamldec":["௦","௧","௨","௩","௪","௫","௬","௭","௮","௯"],"telu":["౦","౧","౨","౩","౪","౫","౬","౭","౮","౯"],"thai":["๐","๑","๒","๓","๔","๕","๖","๗","๘","๙"],"tibt":["༠","༡","༢","༣","༤","༥","༦","༧","༨","༩"],"hanidec":["〇","一","二","三","四","五","六","七","八","九"]}
```
Comment (longlho) [inline packages/intl-numberformat/src/data/digit-mapping.json:1]: can u add `@generated` tag & `prettier-ignore` here?

**Review by longlho — APPROVED** [review]

**Comment by longlho** [comment]
@pyrocat101 reminder: can u wrap all the Object.defineProperty stuff to pass test262 tests in try/catch
older browser has a Cannot redefine property issue

File: packages/intl-numberformat/src/data/currency-digits.json (hunk 1 above)
Comment (pyrocat101) [reply packages/intl-numberformat/src/data/currency-digits.json:1]: This is a JSON file. I don't think I can do that.

File: packages/intl-numberformat/src/data/digit-mapping.json (hunk 2 above)
Comment (pyrocat101) [reply packages/intl-numberformat/src/data/digit-mapping.json:1]: ditto

File: packages/intl-numberformat/src/data/currency-digits.json (hunk 1 above)
Comment (pyrocat101) [reply packages/intl-numberformat/src/data/currency-digits.json:1]: I will use a `.gitattributes` file to mark them as generated in Github.

**Pull request merged**